### ISSUE TITLE
Enhance 'Output pin' behaviour

### ISSUE THREAD

**Issue by willkaxxx** [Clon1998/mobileraker#70]
* Currently for output button we have got slider, even if it have only two positions
![image](https://user-images.githubusercontent.com/48694879/185787457-6ec618df-f465-4c8f-9507-19aa8e8155b0.png)
![image](https://user-images.githubusercontent.com/48694879/185787667-310530d9-53ce-4a6e-b8d7-16a311939d2b.png)

* Could you please add ordinary button (on /off) , if output pin can take only two values?

* Example from Fluidd
![image](https://user-images.githubusercontent.com/48694879/185787556-d3b8021c-77c2-4778-a56c-862a31a04740.png)

Thanks!)

**Comment by Clon1998** [comment]
Adding such an option should not be a problem. I'll add it once I am done migrating to a new framework.

**Comment by Fl0ken** [comment]
Adding to this: could the slider actually enable values between 0 and 1? I got a Daylight on the pin and can only do off/max brightness instead of changing the brightness.

**Comment by Clon1998** [comment]
> Adding to this: could the slider actually enable values between 0 and 1? I got a Daylight on the pin and can only do off/max brightness instead of changing the brightness.

Hey, this depends on the output pin `scale` configuration. Make sure you set the configuration correctly to enable this!

**Comment by Clon1998** [comment]
@willkaxxx sorry for my WAY to late reply, I just went trough the issues and was wondering why this ticket is not closed yet since I implemented the logic for this ages ago. Turns out the config parsing for the `output_pin` was not working correctly causing the app to still use a slider instead of the new switch cards!

I included the fix in the upcoming version `2.3.0`!

**Comment by Clon1998** [comment]
Added in [2.3.0](https://github.com/Clon1998/mobileraker/releases/tag/android2.3.0)

**Comment by Fl0ken** [comment]
> > Adding to this: could the slider actually enable values between 0 and 1? I got a Daylight on the pin and can only do off/max brightness instead of changing the brightness.
> 
> 
> 
> Hey, this depends on the output pin `scale` configuration. Make sure you set the configuration correctly to enable this!

Hey, will try the scale when home, but shouldn't this also work if I got PWM enabled? I can already change the brightness in Mainsail.

**Comment by willkaxxx** [comment]
Hey, @Clon1998 , I have tried to reinstal app, clean cashes, delete printer from configs.
But it still a slider :(
Could you suggest something to try more?
My current version is 2.3.1
My config in klipper like: 
![image](https://github.com/Clon1998/mobileraker/assets/48694879/f576af5f-0d5c-4ca4-bcc5-8afac1375254)

Thanks

**Comment by Clon1998** [comment]
> > > Adding to this: could the slider actually enable values between 0 and 1? I got a Daylight on the pin and can only do off/max brightness instead of changing the brightness.
> > 
> > 
> > 
> > Hey, this depends on the output pin `scale` configuration. Make sure you set the configuration correctly to enable this!
> 
> Hey, will try ```scale``` when home, but shouldn't this also work if I got PWM enabled? I can already change the brightness in Mainsail.

Good point, I will make sure that non pwm pins are always toggles/switches.

**Comment by Clon1998** [comment]
> Hey, @Clon1998 , I have tried to reinstal app, clean cashes, delete printer from configs.
> 
> But it still a slider :(
> 
> Could you suggest something to try more?
> 
> My current version is 2.3.1
> 
> My config in klipper like: 
> 
> ![image](https://github.com/Clon1998/mobileraker/assets/48694879/f576af5f-0d5c-4ca4-bcc5-8afac1375254)
> 
> 
> 
> Thanks

Hey,
I will check that. The pin I tested that with is defined like that
![image](https://github.com/Clon1998/mobileraker/assets/10357775/f051e9ee-7959-4202-87bc-2150158f4768)

**Comment by Clon1998** [comment]
@willkaxxx, I just had another look and I simply can not reproduce this on my device (version 2.3.4).


See my test config:
<img width="239" alt="image" src="https://github.com/Clon1998/mobileraker/assets/10357775/b3962381-1a9e-4966-a15d-fcc5fb0b3d29">

**Comment by willkaxxx** [comment]
@Clon1998 I have found the root cause, if name of pin is "LED" (upper case), it doesnt work as expected, but if I change naming to "led" (lower case), it works fine

**Comment by Clon1998** [comment]
> @Clon1998 I have found the root cause, if name of pin is "LED" (upper case), it doesnt work as expected, but if I change naming to "led" (lower case), it works fine

That's interesting and definitely a Bug in mobileraker. I will double check my implementation.
Thanks for your reply!

**Comment by Clon1998** [comment]
I forgot to close this issue. The parsing for most klipper objects was reworked and the keys/config keys are now case-insensitive.
